test/units/modules/packaging/language/test_pip.py

```diff
@@ -1,23 +1,18 @@
-import json
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-from ansible.compat.tests import unittest
-from ansible.compat.tests.mock import patch
-from ansible.module_utils import basic
+import pytest
 
 from ansible.modules.packaging.language import pip
 
-from units.modules.utils import set_module_args, AnsibleFailJson, ModuleTestCase
 
+pytestmark = pytest.mark.usefixtures('patch_ansible_module')
 
-class TestPip(ModuleTestCase):
-    def setUp(self):
-        super(TestPip, self).setUp()
 
-    @patch.object(basic.AnsibleModule, 'get_bin_path')
-    def test_failure_when_pip_absent(self, mock_get_bin_path):
+@pytest.mark.parametrize('patch_ansible_module', [{'name': 'six'}], indirect=['patch_ansible_module'])
+def test_failure_when_pip_absent(mocker):
+    get_bin_path = mocker.patch('ansible.module_utils.basic.AnsibleModule.get_bin_path')
+    get_bin_path.return_value = None
 
-        mock_get_bin_path.return_value = None
-
-        with self.assertRaises(AnsibleFailJson):
-            set_module_args({'name': 'six'})
-            pip.main()
+    with pytest.raises(SystemExit):
```

Comment:
Both [`exit_json`](https://github.com/ansible/ansible/blob/ebd08d2a01d96d41f8bc1ccf27bc1bbf91060a44/lib/ansible/module_utils/basic.py#L2254) and [`fail_json`](https://github.com/ansible/ansible/blob/ebd08d2a01d96d41f8bc1ccf27bc1bbf91060a44/lib/ansible/module_utils/basic.py#L2261) methods use `sys.exit`: the old code checked that `fail_json` has been called and would have failed if `exit_json` were called in `pip.main`.